packages/edit-post/src/components/header/index.js

```diff
@@ -79,7 +79,7 @@ function Header( { setEntitiesSavedStatesCallback } ) {
 			hasActiveMetaboxes: select( editPostStore ).hasMetaBoxes(),
 			hasHistory: !! select( editorStore ).getEditorSettings().goBack,
 			isEditingTemplate:
-				select( editorStore ).getRenderingMode() === 'template-only',
```

Comment:
A lot of the uses of `isEditingTemplate` in this file seem mostly about adapting how `TopToolbar` works when there's a `<DocumentBar>` being rendered in the middle of a header (as there is when viewing a template or pattern).

It seems like it's not working correctly when viewing a pattern at the moment as the 'show/hide block tools' button isn't present, and so there's a lot of stuff squished in:
![Screenshot 2024-01-10 at 3 37 06 pm](https://github.com/WordPress/gutenberg/assets/677833/b7727a0a-00b4-44ee-a602-fda50b02ea80)

If you compare it to how it works when viewing a template, the block tools can be collapsed and expanded and that also shows and hides the document bar:
![Screenshot 2024-01-10 at 3 39 47 pm](https://github.com/WordPress/gutenberg/assets/677833/96e5aa83-6081-4251-baa2-56bfd5784257)

So I think retaining `isEditingTemplate` isn't quite right, the logic should be more about `hasHistory`.

Reply:
Woops, my bad, fixed.